.github/workflows/ci.yml

```diff
@@ -364,10 +384,34 @@ jobs:
         run: chmod +x ./uv
 
       - name: "Print Python path"
-        run: echo $(which python3.11)
+        run: echo $(which python3)
+
+      - name: "Validate global Python install"
+        run: python3 scripts/check_system_python.py --uv ./uv
+
+  system-test-macos-x86_64:
+    needs: build-binary-macos-x86_64
+    name: "check system | python on macos x86_64"
+    runs-on: macos-14
```

Comment:
It looks like the failing job here ran on macOS 12.7.3: https://github.com/AlexWaygood/typeshed-stats/actions/runs/8279567587/job/22654258397?pr=206

Reply:
I just use "macos-latest" in that job, but it looks like that does not in fact resolve to the latest macos version

Reply:
Oh we should be using `macos-latest` here 🤦

Reply:
(I'm a dumbo)

Reply:
`macos-14` is the ARM machine